hooks/tk-multi-publish2/basic/collector.py

```diff
@@ -0,0 +1,330 @@
+        :param parent_item: Root item instance
+
+        """
+
+        # create an item representing the current maya session
+        item = self.collect_current_maya_session(settings, parent_item)
+        project_root = item.properties["project_root"]
+
+        # look at the render layers to find rendered images on disk
+        self.collect_rendered_images(item)
+
+        # if we can determine a project root, collect other files to publish
+        if project_root:
+
+            self.logger.info(
+                "Current Maya project is: %s." % (project_root,),
+                extra={
+                    "action_button": {
+                        "label": "Change Project",
+                        "tooltip": "Change to a different Maya project",
+                        "callback": lambda: mel.eval('setProject ""')
+                    }
+                }
+            )
+
+            self.collect_playblasts(item, project_root)
+            self.collect_alembic_caches(item, project_root)
+
+        else:
+
+            self.logger.warning(
+                "Could not determine the current Maya project.",
+                extra={
+                    "action_button": {
+                        "label": "Set Project",
+                        "tooltip": "Set the Maya project",
+                        "callback": lambda: mel.eval('setProject ""')
+                    }
+                }
+            )
+
+    def collect_current_maya_session(self, settings, parent_item):
+        """
+        Creates an item that represents the current maya session.
+
+        :param parent_item: Parent Item instance
+
+        :returns: Item of type maya.session
+        """
+
+        publisher = self.parent
+
+        # get the path to the current file
+        path = cmds.file(query=True, sn=True)
+
+        # determine the display name for the item
+        if path:
+            file_info = publisher.util.get_file_path_components(path)
+            display_name = file_info["filename"]
+        else:
+            display_name = "Current Maya Session"
+
+        # create the session item for the publish hierarchy
+        session_item = parent_item.create_item(
+            "maya.session",
+            "Maya Session",
+            display_name
+        )
+
+        # get the icon path to display for this item
+        icon_path = os.path.join(
+            self.parent.engine.disk_location,
+            "resources",
+            "maya.png"
+        )
+        session_item.set_icon_from_path(icon_path)
+
+        # discover the project root which helps in discovery of other
+        # publishable items
+        project_root = cmds.workspace(q=True, rootDirectory=True)
+        session_item.properties["project_root"] = project_root
+
+        # if a work file template is defined, add it to the item properties so
+        # that it can be used by attached publish plugins
+        work_template_setting = settings.get("Work file Template")
```

Comment:
This is the new section here. All it's doing is attaching the configured work file template to the session item. As noted, I realized when doing the implementation that we can't actually evaluate the fields at this point because the path may change... ex: user manually saves with the publisher open or a version number is inserted when the "begin file versioning" plugin runs. This allows the work template to be configured once for the publish session and shared by the publish plugins to use as needed. 

@manneohrstrom brought up the idea of having app-level template configuration and allowing the plugins to pull in what they need from that. I think that could work, but it separates the requirements of a plugin from the configuration which I'm not totally sold on. Anyway, something to consider.